packages/admin-ui/ui/src/components/organisms/product-variants-section/index.tsx

```diff
@@ -71,8 +79,23 @@ const ProductVariantsSection = ({ product }: Props) => {
 
   const { onDeleteVariant } = useEditProductActions(product.id)
 
-  const handleDeleteVariant = (variantId: string) => {
-    onDeleteVariant(variantId)
+  const handleDeleteVariant = async (variantId: string) => {
+    let variantInventory: VariantInventory | undefined
+    if (isFeatureEnabled("inventoryService")) {
+      const { variant } = await client.admin.variants.getInventory(variantId)
+      variantInventory = variant
+    }
+    onDeleteVariant(variantId, async () => {
+      if (
+        isFeatureEnabled("inventoryService") &&
+        variantInventory?.inventory[0]?.id
+      ) {
+        await client.admin.inventoryItems.delete(
+          variantInventory.inventory[0].id
+        )
+        queryClient.invalidateQueries(adminInventoryItemsKeys.lists())
+      }
+    })
```

Comment:
question: What's the reason for not having this live on the backend?

Reply:
Generally we have handled things related to inventory items separately from variants in the admin system. 
We could have this on the backend instead. I added it here to not couple the deletion of the variant with the deletion of the inventory item in the backend since they can exist independently of each other (we cascade delete the association). We could delete the inventory item it in the backend in this first iteration and then remove the deletion at a later time when we expand the capabilities of the MW feature.

wdyt @srindom?

Reply:
I am open to both approaches. 

At this time, it might be more correct to delete the inventory item on the backend (since we are assuming 1:1 relationship between variant and inventory item). 

However, when we later allow multiple inventory items to be associated with multiple variants this logic will be more difficult to have in the backend. E.g., consider a scenario where Item A is associated both with Variant X and Variant Y, should Item A be deleted when Variant X is deleted? Should it be deleted when Variant Y is deleted? Should it be deleted if both are deleted? 

I think there could be merit to the idea of treating inventory items as completely separate entities to the variants. I.e. store operators can create/manage/delete inventory items without going through a particular variant association. But this is all future stuff - for right now I am fine with either approach. Although, I could imagine that moving this to the backend would have to be reversed eventually in the future.

Reply:
Alright, keeping it as is for now 👍